src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/AbstractGenerateNuGetPackageCommand.cs

```diff
@@ -132,20 +132,36 @@ public int OnActiveProjectCfgChange(IVsHierarchy pIVsHierarchy)
         #endregion
 
         #region IDisposable
-        public void Dispose()
+        private bool _disposedValue = false;
+
+        protected virtual void Dispose(bool disposing)
         {
-            // Build manager APIs require UI thread access.
-            _threadingService.ExecuteSynchronously(async() =>
+            if (!_disposedValue)
             {
-                await _threadingService.SwitchToUIThread();
-
-                if (_buildManager != null)
+                if (disposing)
                 {
-                    // Unregister solution build events.
-                    _buildManager.UnadviseUpdateSolutionEvents(_solutionEventsCookie);
-                    _buildManager = null;
+                    // Build manager APIs require UI thread access.
+                    _threadingService.ExecuteSynchronously(async () =>
+                    {
+                        await _threadingService.SwitchToUIThread();
+
+                        if (_buildManager != null)
+                        {
+                            // Unregister solution build events.
+                            _buildManager.UnadviseUpdateSolutionEvents(_solutionEventsCookie);
+                            _buildManager = null;
+                        }
+                    });
                 }
-            });
+
+                _disposedValue = true;
+            }
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
Does this type even have a finalizer to suppress?

Reply:
➡️ This is the form of `Dispose` whether or not the type has a finalizer. If it does not have a finalizer, this will NOP.